chronos/asyncloop.nim

```diff
@@ -180,7 +180,7 @@ type
   CallbackFunc* = proc (arg: pointer = nil) {.gcsafe.}
   CallSoonProc* = proc (c: CallbackFunc, u: pointer = nil) {.gcsafe.}
 
-  AsyncCallback* = object
+  AsyncCallback* = ref object
```

Comment:
What the reason of doing such here?

Reply:
First, this seems like it would reduce the amount of copying required, second I seem to remember you suggesting we do that?

Reply:
This also introduce one more allocation for every possible callback in the `chronos`. And i never suggested such thing, sorry.

Reply:
> This also introduce one more allocation for every possible callback in the chronos
Does it, I'm not sure where? (genuine question)